stdlib/public/TensorFlow/Gradients.swift

```diff
@@ -635,3 +635,67 @@ func _vjpRelu<T : TensorFlowFloatingPoint>(
 ) -> (Tensor<T>, (Tensor<T>) -> Tensor<T>) {
   return (relu(x), { v in Tensor(x .> 0) * v })
 }
+
+//===----------------------------------------------------------------------===//
+// Broadcasting
+//===----------------------------------------------------------------------===//
+
+extension Tensor where Scalar : TensorFlowFloatingPoint {
+  @inlinable
+  func _vjpBroadcast(
+    toShape shape: Tensor<Int32>
+  ) -> (Tensor, (Tensor) -> Tensor) {
+    return (broadcast(toShape: shape), { [origShape = self.shapeTensor] v in
+      v.unbroadcast(toShape: origShape)
+    })
+  }
+  
+  @inlinable
+  func _vjpBroadcast(
+    to shape: TensorShape
+  ) -> (Tensor, (Tensor) -> Tensor) {
+    return (broadcast(to: shape), { [origShape = self.shape] v in
+      v.unbroadcast(to: origShape)
+    })
+  }
+  
+  @inlinable
+  func _vjpBroadcast<OtherScalar>(
+    like other: Tensor<OtherScalar>
+  ) -> (Tensor, (Tensor) -> Tensor)
+    where OtherScalar : TensorFlowScalar {
+    return (broadcast(like: other), { [origShape = self.shapeTensor] v in
+      v.unbroadcast(like: origShape)
```

Comment:
Though this VJP is not necessary as explained in another comment down below, this `broadcast(like: origShape)` is an incorrect implementation of this. `origShape` is a shape, not a tensor value that was broadcast from. The implementation should be either `broadcast(like: self)` or `broadcast(toShape: origShape)`.

Reply:
Ahh yes same mistake as I made yesterday it seems! Didn't catch this one. 😄

Reply:
I'm a bit confused as to what's wrong with the implementation, I'll ask you offline about this! 😄